metaseq/cli/train.py

```diff
@@ -510,28 +521,59 @@ def post_checkpoint_callback(cfg, filename):
                 logger.info(f"could not upload {filename}: {e}")
 
 
-def _run_evaluations(
-    eval_module, cloud_upload_path, local_file, checkpoint_suffix, gloo_pg
+def nfs_evaluation(
+    cfg, num_updates, should_stop, checkpoint_dir, destination_checkpoints_dir
 ):
-    # Make sure all ranks have finished uploading checkpoints.
-    # If any rank doesn't hit the barrier within the timeout period, we throw an error and do
-    # not run evals. Error doesn't stop training run.
-    # dist.monitored_barrier(group=gloo_pg, timeout=timedelta(minutes=5))
-    # Run evals on rank 0
-    if distributed_utils.get_global_rank() != 0:
-        return
-    assert eval_module is not None, "--eval-module needs to be set."
-    module = importlib.import_module(eval_module)
-    if not hasattr(module, "eval_fn"):
-        raise RuntimeError(
-            f"{eval_module} must have a function called eval_fn to utilize for evaluations. "
-            "It expects the following signature:\n"
-            "def eval_fn(cloud_upload_path: str, checkpoint_name: str)"
+    if (
+        cfg.checkpoint.nfs_eval_script_path is not None
+        and distributed_utils.get_global_rank() == 0
+        and cfg.checkpoint.nfs_eval_frequency > 0
+        and num_updates % cfg.checkpoint.nfs_eval_frequency == 0
```

Comment:
Did you mean to use `should_stop` here as an additional check? 

I find it weird not having a flag to control whether we run at the end, are you just going to turn it on by default then? It seems clear to me that we would want to evaluate last checkpoint

cc @suchenzang Peter mentioned you argued in favour of not having flags for last_checkpoint?

Reply:
yeah hopefully no flag and default to just checkpoint the last checkpoint (which we should be doing now since we checkpoint every epoch)? is there a case where the last step is not also the end of an epoch/data shard?

Reply:
- Added option 1: eval last checkpoint by default, assuming nfs_eval_script_path is set.
- renamed as suggested to training_finished, that makes it clearer!